sdk/go.mod

```diff
@@ -0,0 +1,25 @@
+module go.opentelemetry.io/auto/sdk
+
+go 1.22.0
+
+toolchain go1.22.6
+
+require (
+	go.opentelemetry.io/collector/pdata v1.14.1
+	go.opentelemetry.io/otel v1.29.0
+	go.opentelemetry.io/otel/trace v1.29.0
```

Comment:
It shouldn't, and in testing it looks to be the case that it doesn't.

Inter-module dependencies are allowed. They are resolved using the go mod strategy. And since this package can never import `go.openetelemetry.io/otel/internal/global` there is a strong guarantee that there will never be a cyclic import.

However, it verifying this, I did see indirect dependencies causing conflicts from `pdata`:

```terminal
go mod tidy in .
go mod tidy in ./bridge/opencensus
go: go.opentelemetry.io/otel/bridge/opencensus imports
	go.opentelemetry.io/otel imports
	go.opentelemetry.io/otel/internal/global imports
	go.opentelemetry.io/auto/sdk imports
	go.opentelemetry.io/collector/pdata/ptrace imports
	go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1 imports
	google.golang.org/grpc/status imports
	google.golang.org/genproto/googleapis/rpc/status: ambiguous import: found package google.golang.org/genproto/googleapis/rpc/status in multiple modules:
	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013 (/home/tyler/go/pkg/mod/google.golang.org/genproto@v0.0.0-20200526211855-cb27e3aa2013/googleapis/rpc/status)
	google.golang.org/genproto/googleapis/rpc v0.0.0-20240903143218-8af14fe29dc1 (/home/tyler/go/pkg/mod/google.golang.org/genproto/googleapis/rpc@v0.0.0-20240903143218-8af14fe29dc1/status)
go: go.opentelemetry.io/otel/bridge/opencensus imports
	go.opentelemetry.io/otel imports
	go.opentelemetry.io/otel/internal/global imports
	go.opentelemetry.io/auto/sdk imports
	go.opentelemetry.io/collector/pdata/ptrace imports
	go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1 imports
	google.golang.org/grpc/codes tested by
	google.golang.org/grpc/codes.test imports
	google.golang.org/genproto/googleapis/rpc/code: ambiguous import: found package google.golang.org/genproto/googleapis/rpc/code in multiple modules:
	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013 (/home/tyler/go/pkg/mod/google.golang.org/genproto@v0.0.0-20200526211855-cb27e3aa2013/googleapis/rpc/code)
	google.golang.org/genproto/googleapis/rpc v0.0.0-20240903143218-8af14fe29dc1 (/home/tyler/go/pkg/mod/google.golang.org/genproto/googleapis/rpc@v0.0.0-20240903143218-8af14fe29dc1/code)
go: go.opentelemetry.io/otel/bridge/opencensus imports
	go.opentelemetry.io/otel imports
	go.opentelemetry.io/otel/internal/global imports
	go.opentelemetry.io/auto/sdk imports
	go.opentelemetry.io/collector/pdata/ptrace imports
	go.opentelemetry.io/collector/pdata/internal/data/protogen/collector/trace/v1 imports
	google.golang.org/grpc/status tested by
	google.golang.org/grpc/status.test imports
	google.golang.org/genproto/googleapis/rpc/errdetails: ambiguous import: found package google.golang.org/genproto/googleapis/rpc/errdetails in multiple modules:
	google.golang.org/genproto v0.0.0-20200526211855-cb27e3aa2013 (/home/tyler/go/pkg/mod/google.golang.org/genproto@v0.0.0-20200526211855-cb27e3aa2013/googleapis/rpc/errdetails)
	google.golang.org/genproto/googleapis/rpc v0.0.0-20240903143218-8af14fe29dc1 (/home/tyler/go/pkg/mod/google.golang.org/genproto/googleapis/rpc@v0.0.0-20240903143218-8af14fe29dc1/errdetails)
```

This is likely only related to the way I was verifying:

- Update `go.opentelemetry.io/otel` to replace all references to `go.opentelemetry.io/auto/sdk` with the local copy (i.e. a go mod `replace` directive)
- Update `go.opentelemetry.io/otel/internal/global` with the dependency on this SDK
- Run `make go-mod-tidy` in opentelemetry-go

I'm still looking to see if I can resolve this conflict and have a clean build. But it does raise the question of if we want to use `pdata` in the long-term. Maybe we should look at using some custom data-structures encoded with `glob`(?).